### PR TITLE
Feature/molecular feature

### PR DESCRIPTION
**Pull request by jduerholt**
This PR adds the basic implementation of a `MolecularInput` feature, which is currently only usable for models but not yet for any optimizations. For optimizations, a MolecularCategoricalInput has to be added.

The descriptor calculations are based on the GAUCHE lib: https://github.com/leojklarner/gauche

### CODE REVIEW

File: tests/bofire/data_models/test_molecular.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+import pandas as pd
+import pytest
+from pandas.testing import assert_series_equal
+
+from bofire.data_models.features.molecular import MolecularInput
+
+CYIPOPT_AVAILABLE = importlib.util.find_spec("cyipopt") is not None
+
+VALID_SMILES = pd.Series(["CC(=O)Oc1ccccc1C(=O)O", "c1ccccc1"])
```
Comment (jkleinekorte) [inline tests/bofire/data_models/test_molecular.py:11]:
```suggestion
VALID_SMILES = pd.Series(["CC(=O)Oc1ccccc1C(=O)O", "c1ccccc1", "[CH3][CH2][OH]", "C-C-O", "OCC", "N[C@](C)(F)C(=O)O"])
```

File: bofire/utils/cheminformatics.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+try:
+    from rdkit.Chem import AllChem, Descriptors, MolFromSmiles  # type: ignore
+except ImportError:
+    warnings.warn(
+        "rdkit not installed, BoFire's cheminformatics utilities cannot be used."
+    )
+from sklearn.feature_extraction.text import CountVectorizer
+
+# This code is based on GAUCHE: https://github.com/leojklarner/gauche/blob/main/gauche/data_featuriser/featurisation.py
+
```
Comment (jkleinekorte) [inline bofire/utils/cheminformatics.py:15]: what about all the backward transformations? Mol2smiles etc

File: bofire/utils/cheminformatics.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    Args:
+        smiles (List[str]): List of smiles
+
+    Returns:
+        np.ndarray: Array holding the fragment information.
+    """
+    # descList[115:] contains fragment-based features only
+    # (https://www.rdkit.org/docs/source/rdkit.Chem.Fragments.html)
+    # Update: in the new RDKit version the indices are [124:]
+    fragments = {d[0]: d[1] for d in Descriptors.descList[124:]}
```
[the remark this answers is not in the document]
Comment (jduerholt) [reply bofire/utils/cheminformatics.py:71]: It is copied from the GAUCHE code, as I want in a some next step include their cheminformatics models which are based on this fragments. The list of fragments can be found here: https://www.rdkit.org/docs/source/rdkit.Chem.Fragments.html

File: bofire/utils/cheminformatics.py (hunk 2 above)
Comment (jduerholt) [reply bofire/utils/cheminformatics.py:15]: Not yet in the focus.

**Review by jkleinekorte — APPROVED** [review]

**Pull request merged**